### PR TITLE
SLE-23248 Document support limitations for 389-DS

### PR DESCRIPTION
**Pull request by sounix000**
### PR creator: Description

Document certain limitations for better organizing official support around 389-DS.

### PR creator: Are there any relevant issues/feature requests?

* Bugzilla: 
* JIRA: https://jira.suse.com/browse/SLE-23428

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [x] SLE 15 SP4/openSUSE Leap 15.5
  - [x] SLE 15 SP4/openSUSE Leap 15.4
  - [x] SLE 15 SP3/openSUSE Leap 15.3
  - [x] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

File: xml/security_ldap_sssd.xml [hunk 1]
```diff
@@ -197,4 +197,23 @@ session:</screen>
     See <xref linkend="cha-security-auth"/> for information on managing the
     sssd.service with systemctl.
   </para>
+
+  <sect2 xml:id="sec-security-ldap-server-sssd-unsupported-password-hash">
+   <title>Unsupported passworded hashes and authentication schemes</title>
+   <para>
+    The following password hash and authentication schemes are not officially supported on &ds389;:
```
Comment (Firstyear) [inline xml/security_ldap_sssd.xml:204]: The following are not supported as configuration values in dse.ldif for the settings nsslapd-rootpwstoragescheme or passwordStorageScheme, or as a value of passwordStorageScheme in account policy objects. Note that database imports that contain these values *are* supported provided nsslapd-enable-upgrade-hash is `on` (defaults to `on`). 

* SHA
* SSHA
* SHA256
* SSHA256
* SHA384
* SSHA384
* SHA512
* SSHA512
* NS-MTA-MD5
* clear
* MD5
* SMD5

**Review by Firstyear — APPROVED** [review]
Thank you!

File: xml/security_ldap_sssd.xml [hunk 2]
```diff
@@ -197,4 +197,35 @@ session:</screen>
     See <xref linkend="cha-security-auth"/> for information on managing the
     sssd.service with systemctl.
   </para>
+
+  <sect2 xml:id="sec-security-ldap-server-sssd-unsupported-password-hash">
+   <title>Unsupported passworded hashes and authentication schemes</title>
```
Comment (dariavladykina) [inline xml/security_ldap_sssd.xml:202]:
```suggestion
   <title>Unsupported password hashes and authentication schemes</title>
```

File: xml/security_ldap_sssd.xml [hunk 3]
```diff
@@ -197,4 +197,35 @@ session:</screen>
...
+    <listitem><para>NS-MTA-MD5</para></listitem>
+    <listitem><para>clear</para></listitem>
+    <listitem><para>MD5</para></listitem>
+    <listitem><para>SMD5</para></listitem>
+   </itemizedlist>
+   <note>
+    <para>
+      Database imports that contain these values are supported provided the
+      <systemitem>nsslapd-enable-upgrade-hash</systemitem> is set to <systemitem>on</systemitem>
+      (defaults to <systemitem>on</systemitem>).
```
Comment (dariavladykina) [inline xml/security_ldap_sssd.xml:227]:
```suggestion
      Database imports that contain these values are supported if
      <systemitem>nsslapd-enable-upgrade-hash</systemitem> is set to <systemitem>on</systemitem>
      (defaults to <systemitem>on</systemitem>).
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some suggestions here. Thanks!

File: xml/security_ldap_modules.xml [hunk 4]
```diff
@@ -99,4 +99,29 @@ memberOf: cn=SERVER_ADMINS,ou=groups,dc=ldap1,dc=com</screen>
  <command>fixup</command> command:
 </para>
 <screen>&prompt.sudo;<command>dsconf LDAP1 plugin memberof fixup -f '(objectClass=*)' dc=LDAP1,dc=COM</command></screen>
+
+<sect2 xml:id="sec-security-ldap-modules-unsupported-plugins">
+ <title>Unsupported plugins on &ds389;</title>
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply xml/security_ldap_modules.xml:104]: Don't make these changes. Plugin is the language specifically used in 389-ds, so consistency to the server and its configs are key.

File: xml/security_ldap_modules.xml [hunk 5]
```diff
@@ -99,4 +99,29 @@ memberOf: cn=SERVER_ADMINS,ou=groups,dc=ldap1,dc=com</screen>
...
+ <para>
+  The following plugins are not officially supported on &ds389;:
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply xml/security_ldap_modules.xml:106]: As above, do not change.

File: xml/security_ldap_modules.xml [hunk 6]
```diff
@@ -99,4 +99,29 @@ memberOf: cn=SERVER_ADMINS,ou=groups,dc=ldap1,dc=com</screen>
...
+ </para>
+ <itemizedlist>
+  <listitem>
+   <para>
+    Distributed Numeric Assignment (DNA) plugin
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply xml/security_ldap_modules.xml:111]: As above, do not change.

File: xml/security_ldap_modules.xml [hunk 7]
```diff
@@ -99,4 +99,29 @@ memberOf: cn=SERVER_ADMINS,ou=groups,dc=ldap1,dc=com</screen>
...
+   </para>
+  </listitem>
+  <listitem>
+   <para>
+    Managed Entries Plugin (MEP)
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply xml/security_ldap_modules.xml:116]: As above, do not change.

File: xml/security_ldap_modules.xml [hunk 8]
```diff
@@ -99,4 +99,29 @@ memberOf: cn=SERVER_ADMINS,ou=groups,dc=ldap1,dc=com</screen>
...
+   </para>
+  </listitem>
+  <listitem>
+   <para>
+    Posix Winsync plugin
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply xml/security_ldap_modules.xml:121]: As above, do not change.

**Review by dariavladykina — APPROVED** [review]

**Comment by sounix000** [comment]
Cherry-picked to maintenance branches:

- SLE15 SP2: https://github.com/SUSE/doc-sle/commit/a78b19c1cc05b7ecbd99eaa4433358da8064fb39
- SLE15 SP3: https://github.com/SUSE/doc-sle/commit/97d5f37eddcd3c906ba4ae4251f1ad296debff81
- SLE15 SP4: https://github.com/SUSE/doc-sle/commit/e3ae6e29964ff551e6fe008f63e740aff5b8cefb
- SLE15 SP5: https://github.com/SUSE/doc-sle/commit/e9646880fc20b9f1e5fa25187c49e6ea7017bceb

**Pull request merged**